Translation/Extractor/File/FormExtractor.php

```diff
@@ -109,86 +109,154 @@ public function enterNode(Node $node)
-                    foreach ($item->value->items as $sitem) {
-                        $this->parseItem($sitem, $domain);
-                    }
-                    continue;
-                }
+        foreach ($node->items as $item) {
+            if (!$item->key instanceof Node\Scalar\String_) {
+                continue;
+            }
 
-                if ('choices' === $item->key->value && !$item->value instanceof Node\Expr\Array_) {
+            if ('translation_domain' === $item->key->value) {
+                if (!$item->value instanceof Node\Scalar\String_) {
                     continue;
                 }
 
-                if ('label' !== $item->key->value && 'empty_value' !== $item->key->value && 'choices' !== $item->key->value && 'invalid_message' !== $item->key->value && 'attr' !== $item->key->value) {
-                    continue;
-                }
+                $domain = $item->value->value;
+            }
+        }
 
-                if ('choices' === $item->key->value) {
+        return $domain;
+    }
 
-                    //Checking for the choice_as_values in the same form item
-                    $choicesAsValues = false;
-                    foreach ($node->items as $choiceItem) {
-                        if ($choiceItem->key !== null && 'choices_as_values' === $choiceItem->key->value) {
-                            $choicesAsValues = ($choiceItem->value->name->parts[0] === 'true');
-                        }
-                    }
-
-                    foreach ($item->value->items as $sitem) {
-                        // If we have a choice as value that differ from the Symfony default strategy
-                        // we should invert the key and the value
-                        if (Kernel::VERSION_ID < 30000 && $choicesAsValues === true || Kernel::VERSION_ID >= 30000) {
-                            $newItem = clone $sitem;
-                            $newItem->key = $sitem->value;
-                            $newItem->value = $sitem->key;
-                            $sitem = $newItem;
-                        }
-                        $this->parseItem($sitem, $domain);
-                    }
-                } elseif ('attr' === $item->key->value && $item->value instanceof Node\Expr\Array_) {
-                    foreach ($item->value->items as $sitem) {
-                        if ('placeholder' == $sitem->key->value) {
-                            $this->parseItem($sitem, $domain);
-                        }
-                        if ('title' == $sitem->key->value) {
-                            $this->parseItem($sitem, $domain);
-                        }
-                    }
-                } elseif ('invalid_message' === $item->key->value) {
-                    $this->parseItem($item, 'validators');
-                } else {
-                    $this->parseItem($item, $domain);
-                }
+    /**
+     * @param Node $item
+     * @param $domain
+     * @return bool
```

Comment:
This function needs documentation. I do not understand what the return value represent. If you parse an empty value node, what does it mean that you return true?